src/main/java/seedu/address/logic/commands/ChangeDebtCommand.java

```diff
@@ -49,12 +54,21 @@ public CommandResult execute(Model model) throws CommandException {
         }
         Person personToEdit = lastShownList.get(index.getZeroBased());
         Debt currentDebt = personToEdit.getDebt();
+
         Person editedPerson;
+        Debt newDebt;
         if (isAdd) {
-            editedPerson = personToEdit.withDebt(Debt.add(currentDebt, debt));
+            newDebt = Debt.add(currentDebt, debt);
+            if (newDebt.compareTo(MAX_DEBT) == 1) {
```

Comment:
Maybe can create a function to abstract the checking out?  E.g. `isDebtOutOfRange(newDebt)`, could be created in `Debt` class as a static method